src/main/java/com/acme/edu/Logger.java

```diff
@@ -2,111 +2,281 @@
 
 public class Logger {
 
+    private enum LastLoggedType {
+        NONE,
+        OBJ,
+        BYTE,
+        INT,
+        CHAR,
+        BOOL,
+        STRING
+    }
+
+    private static LastLoggedType lastLoggedType = LastLoggedType.NONE;
+
     public static final String PRIMITIVE_PREFIX = "primitive: ";
     public static final String CHAR_PREFIX = "char: ";
     public static final String STRING_PREFIX = "string: ";
     public static final String OBJECT_PREFIX = "reference: ";
 
     private static int intBuffer;
-    private static int intMaxMinValueCounter;
-    private static boolean lastLogIntegerFlag;
+    private static int integerThresholdValueCounter;
+
+    private static int byteBuffer;
+    private static int byteThresholdValueCounter;
 
+    private static int repeatableStringCounter;
+    private static String lastString;
+
+    //region loggers
     public static void log(int message) {
-        //save(formatMessage(PRIMITIVE_PREFIX, message));
         checkIntOverFlow(message);
-        lastLogIntegerFlag = true;
+
+        if (!lastLoggedType.equals(LastLoggedType.INT)) {
+            processAllBuffers();
+        }
+
+        lastLoggedType = LastLoggedType.INT;
+    }
+
+    public static void log(byte message) {
+        checkByteOverFlow(message);
+
+        if (!lastLoggedType.equals(LastLoggedType.BYTE)) {
+            processAllBuffers();
+        }
+
+        lastLoggedType = LastLoggedType.BYTE;
     }
 
-    private static  void checkIntOverFlow(int message) {
+    public static void log(char message) {
+        processAllBuffers();
+        save(formatMessage(CHAR_PREFIX, message));
+        lastLoggedType = LastLoggedType.CHAR;
+    }
+
+    public static void log(boolean message) {
+        processAllBuffers();
+        save(formatMessage(PRIMITIVE_PREFIX, message));
+        lastLoggedType = LastLoggedType.BOOL;
+    }
+
+    public static void log(String message) {
+        if (!lastLoggedType.equals(LastLoggedType.STRING) || !message.equals(lastString)) {
+            processAllBuffers();
+        }
+
+        lastString = message;
+        ++repeatableStringCounter;
+
+        lastLoggedType = LastLoggedType.STRING;
+    }
+
+    public static void log(Object message) {
+        processAllBuffers();
+        save(formatMessage(OBJECT_PREFIX, message));
+        lastLoggedType = LastLoggedType.OBJ;
+    }
+    //endregion
+
+    //region int methods
+    private static void checkIntOverFlow(int message) {
+        if (message == Integer.MAX_VALUE) {
+            increaseIntegerThresholdValueCounter();
+            return;
+        }
```

Comment:
Isn't it better to put this specific case into general checkMaxIntOverFlow() method?